pulp_ostree/app/viewsets.py

```diff
@@ -173,6 +86,36 @@ def sync(self, request, pk):
         )
         return core.OperationPostponedResponse(result, request)
 
+    @extend_schema(
+        description="Trigger an asynchronous task to create a new OSTree repository version.",
+        summary="Create a new OSTree repository version",
+        responses={202: AsyncOperationResponseSerializer},
+    )
+    @action(detail=True, methods=["post"], serializer_class=serializers.OstreeRepoUploadSerializer)
+    def upload(self, request, pk):
+        """Upload and parse a tarball consisting of one or more OSTree commits."""
+        repository = self.get_object()
+        request.data["repository"] = repository
+
+        serializer = serializers.OstreeRepoUploadSerializer(
+            data=request.data, context={"request": request}
+        )
+        serializer.is_valid(raise_exception=True)
+
+        artifact = serializer.validated_data["artifact"]
+        repository_name = serializer.validated_data["repository_name"]
+
+        async_result = dispatch(
+            tasks.parse_tarball,
```

Comment:
Yes, it parses the tarball, but it does way more than that. How about `import_ostree_repo`?